js_modules/dagit/packages/core/src/app/ExecutionSessionStorage.tsx

```diff
@@ -176,9 +176,9 @@ export function useExecutionSessionStorage(
 
   // TODO: Remove this migration logic in a few patches when we know the old namespace is likely no longer being used
   const oldDataMigrated = React.useRef(false);
-  if (!oldDataMigrated.current) {
+  if (oldData && !oldDataMigrated.current) {
     onSave(oldData);
-    window.localStorage.removeItem(oldNamespace);
+    window.localStorage.removeItem(getKey(oldNamespace));
     oldDataMigrated.current = true;
   }
```

Comment:
It occurs to me that we should maybe be doing side-effectful stuff like this in an effect instead.

Reply:
Yeah we can. The only downside is it'll be an extra render until the data shows up

Reply:
https://github.com/dagster-io/dagster/pull/7540/files